astroid/nodes/node_classes.py

```diff
@@ -1527,55 +1527,31 @@ class Comprehension(NodeNG):
     _astroid_fields = ("target", "iter", "ifs")
     _other_fields = ("is_async",)
 
-    optional_assign = True
-    """Whether this node optionally assigns a variable."""
-
-    lineno: None
-    col_offset: None
-    end_lineno: None
-    end_col_offset: None
-
-    def __init__(self, parent: NodeNG | None = None) -> None:
-        """
-        :param parent: The parent node in the syntax tree.
-        """
-        self.target: NodeNG | None = None
-        """What is assigned to by the comprehension."""
+    target: NodeNG
+    """What is assigned to by the comprehension."""
 
-        self.iter: NodeNG | None = None
-        """What is iterated over by the comprehension."""
+    iter: NodeNG
+    """What is iterated over by the comprehension."""
 
-        self.ifs: list[NodeNG] = []
-        """The contents of any if statements that filter the comprehension."""
+    ifs: list[NodeNG]
+    """The contents of any if statements that filter the comprehension."""
 
-        self.is_async: bool | None = None
-        """Whether this is an asynchronous comprehension or not."""
+    is_async: bool
+    """Whether this is an asynchronous comprehension or not."""
 
-        super().__init__(parent=parent)
+    optional_assign = True
```

Comment:
Same comment as about `_astroid_fields`, no need to move this in this PR 😄

Reply:
Might as well keep it consistent.